src/lower/merge_lattice.cpp

```diff
@@ -784,6 +857,21 @@ class MergeLatticeBuilder : public IndexNotationVisitorStrict, public IterationA
     return deduplicates;
   }
 
+  static vector<Iterator>
+  removeDimensionIterators(const vector<Iterator>& iterators)
+  {
+    vector<Iterator> result;
+
+    // Remove all but one of the dense iterators, which are all the same.
```

Comment:
We are removing all of the dimension iterators right? Not all but one?

Reply:
Yea. This was a comment I copied from your `deduplicateDImensionIterators(...)` function. This comment is wrong, but the code should remove all dense iterators. I will fix this in the same commit when I change the code to `taco::util::contains`